src/backend/web/graphql/index.js

```diff
@@ -84,18 +112,57 @@ module.exports.resolvers = {
      * @param perPage Number of Post objects in every page
      * @return Array of 'perPage' number of Post objects
      */
-    getPosts: async (parent, { page, perPage }) => {
+    getPosts: async (parent, { filter, page, perPage }) => {
+      let postIds;
+      const numOfPosts = await getPostsCount();
       const prPage = perPage > maxPostsPerPage ? maxPostsPerPage : perPage;
 
-      const numOFPosts = await getPostsCount();
+      // For Date Filter
+      if (filter) {
+        if (filter.fromDate || filter.toDate) {
+          const fromDate = filter.fromDate ? filter.fromDate : new Date();
+          const toDate = filter.toDate ? filter.toDate : new Date();
+          postIds = await getPostsByDate(fromDate, toDate); // Get all results within the date
+          // check if # of filtered results are less than max results allowed per page.
+          if (postIds.length < prPage) {
+            return Promise.all(postIds.map(id => Post.byId(id))); // return all the posts' info
+          }
+          // Otherwise do some pagination here
+          const pageResult = postIds.slice(page - 1 * perPage, page * prPage);
+          return Promise.all(pageResult.map(id => Post.byId(id)));
+        }
+        // Other filters use the getPosts() instead of getPostsByDate()
+        postIds = await getPosts(0, numOfPosts);
+        const result = await Promise.all(postIds.map(id => Post.byId(id)));
+
+        if (filter.url) {
```

Comment:
I think you have `if(filter.url)` here (but this deals with authors, and `if(filter.author)` below this where you deal with urls.